api/pkg/service/auth/auth.go

```diff
@@ -81,7 +82,26 @@ func (r *request) authenticate(code string) (*auth.AuthenticateResult, error) {
 
 	// gets user details from github using the access_token
 	oauthClient := r.oauth.Client(context.Background(), token)
-	ghClient := github.NewClient(oauthClient)
+
+	var ghClient *github.Client
+
+	if !strings.HasPrefix(app.GitHubUrl, "https://github.com") {
+		parsedUrl, err := url.Parse(app.GitHubUrl)
+		if err != nil {
+			return nil, fmt.Errorf("There was some problem while parsing the error")
+		}
+
+		apiUrl := fmt.Sprintf("%s://api.%s", parsedUrl.Scheme, parsedUrl.Host)         // https://api.myghe.com
+		uploadsUrl := fmt.Sprintf("%s://uploads.%s", parsedUrl.Scheme, parsedUrl.Host) // https://uploads.myghe.com
+
+		ghClient, err = github.NewEnterpriseClient(apiUrl, uploadsUrl, oauthClient)
+		if err != nil {
+			return nil, err
+		}
+	} else {
```

Comment:
I think we can remove `else`

Reply:
nope because on the basis of url we provide client is going to be initialised